src/bytes.rs

```diff
@@ -1401,6 +1401,15 @@ impl BytesMut {
     pub fn unsplit(&mut self, other: BytesMut) {
         let ptr;
 
+        if other.inner.len == 0 {
```

Comment:
You cannot access the fields directly as they may be used to store the inline data. Instead, use the fns.